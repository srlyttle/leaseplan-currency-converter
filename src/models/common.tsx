export interface CurrencyResponse {
  rates: Object;
  base: string;
  date: string;
}

export interface CurrencyProps {
  // valueChanged: (event: React.ChangeEvent<HTMLInputElement>) => void;
  //removeCurrency: (event: React.MouseEvent<HTMLInputElement>) => void;
  currencyDetails: CurrencyDetails;
}
export interface CurrencyItemProps {
  removeCurrency: (event: React.MouseEvent<HTMLInputElement>) => void;
  currencyDetails: CurrencyDetails;
  currencyValue: string;
}

export interface CurrencyListProps {
  currencies: CurrencyResponse;
}
export interface CurrencyDetails {
  valueToBase: string;
  isBaseCurrency: boolean;
  currencyCode: string;
  currencyFlag: string;
  currencySymbol: string;
  currencyName: string;
  currencySummary: string;
  latestRate: number;
  lastDate?: string;
  baseCalcCurrency?: string;
  displayForUser: boolean;
}
