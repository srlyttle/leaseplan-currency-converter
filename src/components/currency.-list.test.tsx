import React from 'react';
import { render } from '@testing-library/react';
import { screen, fireEvent } from '@testing-library/dom';
import { CurrencyList } from './currency-list.component';
import { CurrencyDetails } from '../models/common';

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };
const data = {
  base: 'EUR',
  date: '2020-07-17',
  rates: {
    AUD: 1.636,
    CAD: 1.551,
    GBP: 0.91078,
    HKD: 8.8617,
  },
};
describe('<CurrencyList/>', () => {
  test('should render correct input for a list of currencies & display input  & select', () => {
    const { getByText, getByTestId, getAllByDisplayValue, container } = render(
      <CurrencyList currencies={data} />
    );
    screen.getByRole('spinbutton');
    //number input
    expect(screen.getByRole('spinbutton')).toBeInTheDocument();
    expect(screen.getByRole('spinbutton').getAttribute('value')).toBe('1');

    const selectElement = screen.getByTestId('testid-currency-select');
    expect(selectElement).toBeInTheDocument();

    expect(screen.getByRole('combobox')).toBeInTheDocument();

    expect(screen.getByRole('combobox')).toBeInTheDocument();

    expect(getByText('1 CAD = 1.551 EUR')).toBeInTheDocument();

    expect(getByText('1 HKD = 8.8617 EUR')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
