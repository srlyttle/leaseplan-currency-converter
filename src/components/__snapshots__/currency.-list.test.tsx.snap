// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyList/> should render correct input for a list of currencies & display input  & select 1`] = `
<div>
  <div
    class="ant-page-header site-page-header ant-page-header-ghost"
  >
    <div
      class="ant-page-header-heading"
    >
      <div
        class="ant-page-header-heading-left"
      >
        <span
          class="ant-page-header-heading-title"
          title="Currency Converter"
        >
          Currency Converter
        </span>
        <span
          class="ant-page-header-heading-sub-title"
          title="Updated daily from https://exchangeratesapi.io/"
        >
          Updated daily from https://exchangeratesapi.io/
        </span>
      </div>
    </div>
  </div>
  <div
    class="base-currency"
  >
    <div
      class="base-value"
    >
      <div
        class="label-text"
      >
         Base Value:
      </div>
      <div
        class="ant-input-number"
      >
        <div
          class="ant-input-number-handler-wrap"
        >
          <span
            aria-label="Increase Value"
            class="ant-input-number-handler ant-input-number-handler-up"
            role="button"
            unselectable="unselectable"
          >
            <span
              aria-label="up"
              class="anticon anticon-up ant-input-number-handler-up-inner"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="up"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                />
              </svg>
            </span>
          </span>
          <span
            aria-disabled="true"
            aria-label="Decrease Value"
            class="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
            role="button"
            unselectable="unselectable"
          >
            <span
              aria-label="down"
              class="anticon anticon-down ant-input-number-handler-down-inner"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          class="ant-input-number-input-wrap"
        >
          <input
            aria-valuemax="1000"
            aria-valuemin="1"
            aria-valuenow="1"
            autocomplete="off"
            class="ant-input-number-input"
            data-testid="testid-currency-input-value"
            max="1000"
            min="1"
            role="spinbutton"
            step="1"
            type="number"
            value="1"
          />
        </div>
      </div>
    </div>
    <div
      class="base-currency-selection"
    >
      <div
        class="label-text"
      >
        Base Currency:
      </div>
      <div
        class="ant-select ant-select-single ant-select-show-arrow"
        data-testid="testid-currency-select"
        style="width: 200px;"
      >
        <div
          class="ant-select-selector"
        >
          <span
            class="ant-select-selection-search"
          >
            <input
              aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
              aria-autocomplete="list"
              aria-controls="rc_select_TEST_OR_SSR_list"
              aria-haspopup="listbox"
              aria-owns="rc_select_TEST_OR_SSR_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              id="rc_select_TEST_OR_SSR"
              readonly=""
              role="combobox"
              style="opacity: 0;"
              unselectable="on"
              value=""
            />
          </span>
          <span
            class="ant-select-selection-item"
            title="Euro"
          >
            Euro
          </span>
        </div>
        <span
          aria-hidden="true"
          class="ant-select-arrow"
          style="user-select: none;"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-suffix"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    class="list-container"
  >
    <div
      class="ant-list ant-list-split"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <ul
            class="ant-list-items"
          >
            <li
              class="ant-list-item"
            >
              <div
                class="ant-list-item-meta"
              >
                <div
                  class="ant-list-item-meta-avatar"
                >
                  <span
                    class="ant-avatar ant-avatar-circle ant-avatar-image"
                  >
                    <img
                      src="http://www.geonames.org/flags/l/au.gif"
                    />
                  </span>
                </div>
                <div
                  class="ant-list-item-meta-content"
                >
                  <h4
                    class="ant-list-item-meta-title"
                  >
                    AUD
                  </h4>
                </div>
              </div>
              <div
                class="currency-card"
                id="AUD"
              >
                <div
                  class="currency-value"
                >
                  <div
                    class="currency-symbol"
                    data-testid="testid-currency-symbol"
                  >
                    $
                  </div>
                  <div
                    class="currency-value-to-base"
                    data-testid="testid-currency-value-AUD"
                  >
                    1.6360
                  </div>
                </div>
                <div
                  class="currency-summary"
                  data-testid="testid-currency-summary"
                >
                  1 AUD = 1.636 EUR
                </div>
              </div>
            </li>
            <li
              class="ant-list-item"
            >
              <div
                class="ant-list-item-meta"
              >
                <div
                  class="ant-list-item-meta-avatar"
                >
                  <span
                    class="ant-avatar ant-avatar-circle ant-avatar-image"
                  >
                    <img
                      src="http://www.geonames.org/flags/l/ca.gif"
                    />
                  </span>
                </div>
                <div
                  class="ant-list-item-meta-content"
                >
                  <h4
                    class="ant-list-item-meta-title"
                  >
                    CAD
                  </h4>
                </div>
              </div>
              <div
                class="currency-card"
                id="CAD"
              >
                <div
                  class="currency-value"
                >
                  <div
                    class="currency-symbol"
                    data-testid="testid-currency-symbol"
                  >
                    $
                  </div>
                  <div
                    class="currency-value-to-base"
                    data-testid="testid-currency-value-CAD"
                  >
                    1.5510
                  </div>
                </div>
                <div
                  class="currency-summary"
                  data-testid="testid-currency-summary"
                >
                  1 CAD = 1.551 EUR
                </div>
              </div>
            </li>
            <li
              class="ant-list-item"
            >
              <div
                class="ant-list-item-meta"
              >
                <div
                  class="ant-list-item-meta-avatar"
                >
                  <span
                    class="ant-avatar ant-avatar-circle ant-avatar-image"
                  >
                    <img
                      src="http://www.geonames.org/flags/l/uk.gif"
                    />
                  </span>
                </div>
                <div
                  class="ant-list-item-meta-content"
                >
                  <h4
                    class="ant-list-item-meta-title"
                  >
                    GBP
                  </h4>
                </div>
              </div>
              <div
                class="currency-card"
                id="GBP"
              >
                <div
                  class="currency-value"
                >
                  <div
                    class="currency-symbol"
                    data-testid="testid-currency-symbol"
                  >
                    £
                  </div>
                  <div
                    class="currency-value-to-base"
                    data-testid="testid-currency-value-GBP"
                  >
                    0.9108
                  </div>
                </div>
                <div
                  class="currency-summary"
                  data-testid="testid-currency-summary"
                >
                  1 GBP = 0.91078 EUR
                </div>
              </div>
            </li>
            <li
              class="ant-list-item"
            >
              <div
                class="ant-list-item-meta"
              >
                <div
                  class="ant-list-item-meta-avatar"
                >
                  <span
                    class="ant-avatar ant-avatar-circle ant-avatar-image"
                  >
                    <img
                      src="http://www.geonames.org/flags/l/hk.gif"
                    />
                  </span>
                </div>
                <div
                  class="ant-list-item-meta-content"
                >
                  <h4
                    class="ant-list-item-meta-title"
                  >
                    HKD
                  </h4>
                </div>
              </div>
              <div
                class="currency-card"
                id="HKD"
              >
                <div
                  class="currency-value"
                >
                  <div
                    class="currency-symbol"
                    data-testid="testid-currency-symbol"
                  >
                    $
                  </div>
                  <div
                    class="currency-value-to-base"
                    data-testid="testid-currency-value-HKD"
                  >
                    8.8617
                  </div>
                </div>
                <div
                  class="currency-summary"
                  data-testid="testid-currency-summary"
                >
                  1 HKD = 8.8617 EUR
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
